apps/examples/testcase/le_tc/kernel/tc_libc_unistd.c

```diff
@@ -241,6 +242,10 @@ static void tc_libc_unistd_pipe(void)
 
 	sem_wait(&pipe_sem);
 
+	/* wait for child to exit, and store child's exit status */
+	ret_chk = wait(&status);
```

Comment:
yes, I also think sem_`wait(&pipe_sem)` should be removed and should add `waitpid(pid[0], ...) `and `waitpid(pid[1], ...)`

and regarding status variable check `DEBUGASSERT` in `os/kernel/sched/sched_waitpid.c +209`